SDK/AppCenter/Microsoft.AppCenter.Windows.Shared/Utils/UserIdContext.cs

```diff
@@ -48,7 +56,27 @@ public static UserIdContext Instance
         /// <summary>
         /// Current user identifier.
         /// </summary>
-        public string UserId { get; set; }
+        public string UserId
+        {
+            get
+            {
+                lock (UserIdContextLock)
+                {
+                    return _userId;
+                }
+            }
+            set
+            {
+                lock (UserIdContextLock)
+                {
+                    if (_userId != value)
+                    {
+                        _userId = value;
+                    }
+                }
+                UserIdUpdated?.Invoke(this, new UserIdUpdatedEventArgs { UserId = value });
```

Comment:
Only invoke event when there is a change, extract the check as a boolean to reuse it as a new if there